lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/StackController.java

```diff
@@ -398,12 +403,11 @@ private void startChildrenBellowTopChild() {
         }
     }
 
-    private void onNavigationButtonPressed(String buttonId) {
-        if (Constants.BACK_BUTTON_ID.equals(buttonId)) {
+    private void onNavigationButtonPressed(ButtonOptions button) {
+        if (button.shouldPopOnPress())
```

Comment:
@yogevbd Shouldn't we check if the button is a back button? `button.isBackButton()` ? Previously we were checking `if (Constants.BACK_BUTTON_ID.equals(buttonId))`

Reply:
Ohhh you pulled the comparison into the `shouldPopOnPress` - pretty confusing IMO